src/News/NearweekSidebarFeed.jsx

```diff
@@ -0,0 +1,259 @@
+const accountId = "near";
+const limit = props.limit || 5;
+let posts = [];
+
+const indexedPosts = Social.index("post", "main", {
+  accountId,
+  limit: limit,
+  order: "desc",
+});
+
+if (indexedPosts?.length > 0) {
+  posts = [];
+
+  indexedPosts.forEach((post) => {
+    const data = Social.get(`${post.accountId}/post/main`, post.blockHeight);
+    if (data) {
+      const json = JSON.parse(data);
+      const content = json.text.split("\n");
+      const title = content[0] || "";
+      const url = content[1] || content[2] || "";
+      const lastLine = content.pop() || "";
+      const hasNewsTag = lastLine.indexOf("#news") > -1;
+      const isValid = hasNewsTag && url.indexOf("https://") > -1;
+
+      if (isValid) {
+        const timeResponse = fetch(`https://api.near.social/time?blockHeight=${post.blockHeight}`);
+        let createdAt = "";
+        if (timeResponse) {
+          const timeMs = parseFloat(timeResponse.body);
+          createdAt = new Date(timeMs).toISOString();
+          posts.push({
+            blockHeight: post.blockHeight,
+            title,
+            url,
+            thumbnail: "https://near.org/favicon.png",
+            createdAt: new Date(timeMs).toISOString(),
+            categories: ["Near ORG", "blog"],
+          });
+          posts.sort((a, b) => b.blockHeight - a.blockHeight);
+        }
+      }
+    }
+  });
+}
+
+const data = fetch("https://nearweek.com/api/md/dao-news?populate=deep&sort=createdAt:desc&pagination[pageSize]=5", {
+  //subscribe: true,
+  method: "GET",
+  headers: {
+    Accept: "*/*",
+    Authorization:
+      "Bearer 15699f0723aa9fe9f655b1a94e450552476c08807f67b525b5a3c8011eecc8aee6d45923443620f17815b897858be058cd7bd89ddf23a28aabaecb178e7ebc55d380293beeb51a8ce87b40e1518ce4708e4d51a06b115f27fa64ab5cbee5a3511cec785d7ae6a155ecd05ac8196aadae3e9b8e9401b8df8d8b69904f7364f925",
```

Comment:
Is this a good idea to post credentials like this?

Reply:
That's a fair concern. In this case, we were simply forking existing code from NEARWEEK that already exposed this authorizaton header: https://near.org/near/widget/ComponentDetailsPage?src=nearweekapp.near/widget/nearweek-news&tab=source

Reply:
Got it, thanks for the info @calebjacob!